src/Adapter/Order/GeneralConfiguration.php

```diff
@@ -26,39 +26,46 @@
+use Symfony\Component\OptionsResolver\OptionsResolver;
 
 /**
  * General Settings configuration available in ShopParameters > Order Preferences.
  */
-class GeneralConfiguration implements DataConfigurationInterface
+class GeneralConfiguration extends AbstractMultistoreConfiguration
 {
     /**
-     * @var Configuration
+     * @var array<int, string>
      */
-    private $configuration;
-
-    public function __construct(Configuration $configuration)
-    {
-        $this->configuration = $configuration;
-    }
+    private const CONFIGURATION_FIELDS = [
+        'enable_final_summary',
+        'enable_guest_checkout',
+        'disable_reordering_option',
+        'purchase_minimum_value',
+        'recalculate_shipping_cost',
+        'allow_multishipping',
+        'allow_delayed_shipping',
+        'enable_tos',
+        'tos_cms_id',
+    ];
 
     /**
      * {@inheritdoc}
      */
     public function getConfiguration()
     {
+        $shopConstraint = $this->getShopConstraint();
+
         return [
-            'enable_final_summary' => $this->configuration->getBoolean('PS_FINAL_SUMMARY_ENABLED'),
-            'enable_guest_checkout' => $this->configuration->getBoolean('PS_GUEST_CHECKOUT_ENABLED'),
-            'disable_reordering_option' => $this->configuration->getBoolean('PS_DISALLOW_HISTORY_REORDERING'),
-            'purchase_minimum_value' => $this->configuration->get('PS_PURCHASE_MINIMUM'),
-            'recalculate_shipping_cost' => $this->configuration->getBoolean('PS_ORDER_RECALCULATE_SHIPPING'),
-            'allow_multishipping' => $this->configuration->getBoolean('PS_ALLOW_MULTISHIPPING'),
-            'allow_delayed_shipping' => $this->configuration->getBoolean('PS_SHIP_WHEN_AVAILABLE'),
-            'enable_tos' => $this->configuration->getBoolean('PS_CONDITIONS'),
-            'tos_cms_id' => $this->configuration->get('PS_CONDITIONS_CMS_ID'),
+            'enable_final_summary' => (bool) $this->configuration->get('PS_FINAL_SUMMARY_ENABLED', false, $shopConstraint),
+            'enable_guest_checkout' => (bool) $this->configuration->get('PS_GUEST_CHECKOUT_ENABLED', false, $shopConstraint),
+            'disable_reordering_option' => (bool) $this->configuration->get('PS_DISALLOW_HISTORY_REORDERING', false, $shopConstraint),
+            'purchase_minimum_value' => $this->configuration->get('PS_PURCHASE_MINIMUM', null, $shopConstraint),
+            'recalculate_shipping_cost' => (bool) $this->configuration->get('PS_ORDER_RECALCULATE_SHIPPING', false, $shopConstraint),
+            'allow_multishipping' => (bool) $this->configuration->get('PS_ALLOW_MULTISHIPPING', false, $shopConstraint),
+            'allow_delayed_shipping' => (bool) $this->configuration->get('PS_SHIP_WHEN_AVAILABLE', false, $shopConstraint),
+            'enable_tos' => (bool) $this->configuration->get('PS_CONDITIONS', false, $shopConstraint),
+            'tos_cms_id' => $this->configuration->get('PS_CONDITIONS_CMS_ID', null, $shopConstraint),
```

Comment:
```suggestion
            'tos_cms_id' => (int) $this->configuration->get('PS_CONDITIONS_CMS_ID', null, $shopConstraint),
```

Reply:
Since you defined the types in the resolver It's probably better to make them match